bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/event/EventCreators.kt

```diff
@@ -157,80 +157,124 @@ internal fun newErrorMetricsEvent(
 ): Event {
-  val metricEventType = error.toMetricEventType()
   return Event(
     id = idGenerator.newId(),
     type = EventType.METRICS,
-    event = EventData.MetricsEvent(
-      timestamp = clock.currentTimeSeconds(),
-      type = metricEventType,
-      event = newErrorMetricsEventData(featureTag, apiId, metricEventType),
-      sdkVersion = BuildConfig.SDK_VERSION,
-      metadata = newMetadata(appVersion),
-    ),
+    event = newEventDataMetricEvent(error, clock.currentTimeSeconds(), featureTag, appVersion, apiId),
   )
 }
-internal fun newErrorMetricsEventData(
+
+internal fun newEventDataMetricEvent(
+  error: BKTException,
+  timestamp: Long,
   featureTag: String,
+  appVersion: String,
   apiId: ApiId,
-  type: MetricsEventType,
-): MetricsEventData {
-  val labels = mapOf("tag" to featureTag)
+): EventData.MetricsEvent {
+  val labels = mutableMapOf("tag" to featureTag)
+  val (type, event) =
+    when (error) {
+      is BKTException.BadRequestException -> Pair(
+        MetricsEventType.BAD_REQUEST_ERROR,
+        MetricsEventData.BadRequestErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.ClientClosedRequestException -> Pair(
+        MetricsEventType.CLIENT_CLOSED_REQUEST_ERROR,
+        MetricsEventData.ClientClosedRequestErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.FeatureNotFoundException -> Pair(
+        MetricsEventType.NOT_FOUND_ERROR,
+        MetricsEventData.NotFoundErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.ForbiddenException -> Pair(
+        MetricsEventType.FORBIDDEN_ERROR,
+        MetricsEventData.ForbiddenErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.IllegalArgumentException,
+      is BKTException.IllegalStateException,
+      is BKTException.InvalidHttpMethodException,
+      -> Pair(
+        MetricsEventType.INTERNAL_SDK_ERROR,
+        MetricsEventData.InternalSdkErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.InternalServerErrorException -> Pair(
+        MetricsEventType.INTERNAL_SERVER_ERROR,
+        MetricsEventData.InternalServerErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.NetworkException -> Pair(
+        MetricsEventType.NETWORK_ERROR,
+        MetricsEventData.NetworkErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.ServiceUnavailableException -> Pair(
+        MetricsEventType.SERVICE_UNAVAILABLE_ERROR,
+        MetricsEventData.ServiceUnavailableErrorMetricsEvent(
+          apiId = apiId,
+          labels = labels,
+        ),
+      )
+
+      is BKTException.TimeoutException -> {
```

Comment:
can you remove these curly brackets?

```suggestion
      is BKTException.TimeoutException -> Pair(...
```

Reply:
sure, I removed it in https://github.com/bucketeer-io/android-client-sdk/pull/91/commits/beedfddb51339fc237cbd113dd247aad17e54a94